ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlRestApplication.java

```diff
@@ -698,8 +700,16 @@ static KsqlRestApplication buildApplication(
     final RoutingFilterFactory routingFilterFactory = initializeRoutingFilterFactory(ksqlConfig,
         heartbeatAgent, lagReportingAgent);
 
+    final Boolean collectMetrics = ksqlConfig.getBoolean(
+        KsqlConfig.KSQL_QUERY_PULL_METRICS_ENABLED);
+    final Optional<PullQueryExecutorMetrics> pullQueryMetrics = collectMetrics
+        ? Optional.of(new PullQueryExecutorMetrics(
+        ksqlEngine.getServiceId(),
+        ksqlConfig.getStringAsMap(KsqlConfig.KSQL_CUSTOM_METRICS_TAGS)))
+        : empty();
+
```

Comment:
Can we move this into the `PullQueryExecutor` constructor please, given its already being passed the `ksqlConfig` as a parameter?

Reply:
Done